### PR TITLE
[RFE] [ActivationKeys] Update tests to be SCA-compliant 

### PR DESCRIPTION
**Pull request by ColeHiggins2**
SCA is now the default in 6.16. This PR is for updating all tests under the Activation Key Component
SAT-24352

### CODE REVIEW

File: tests/foreman/ui/test_activationkey.py [hunk 1]
```diff
@@ -967,17 +960,13 @@ def test_positive_add_multiple_aks_to_system(session, module_org, rhel6_contenth
...
-                for subscription in ak['subscriptions']['resources']['assigned']
-            ]
-            assert product_name in subscriptions
+            ak = session.activationkey.read(key_name, widget_names='repository sets')[
+                'repository sets'
+            ]['table'][0]
+            assert product_name == ak['Product Name']
         # Create VM
         rhel6_contenthost.install_katello_ca(target_sat)
         rhel6_contenthost.register_contenthost(module_org.label, ','.join([key_1_name, key_2_name]))
```
Comment (vsedmik) [inline tests/foreman/ui/test_activationkey.py:969]: I think we could flip this to the global registration within scope of this PR (it is used on 8 other places in this module).
Also, IIRC RHEL6 support will be dropped this summer and I believe that PRT will fail to checkout RHEL6. We should rather proceed with `rhel_contenthost` and parametrize the test for RHEL8 or 9. And f-string.

File: tests/foreman/cli/test_activationkey.py [hunk 2]
```diff
@@ -1520,42 +1419,37 @@ def test_positive_view_subscriptions_by_non_admin_user(
...
-    activation_key = module_target_sat.cli_factory.make_activation_key(
-        {'name': ak_name, 'organization-id': org.id}
-    )
-    module_target_sat.cli.ActivationKey.add_subscription(
-        {'id': activation_key['id'], 'subscription-id': subscription_id}
+    result = module_target_sat.cli_factory.setup_org_for_a_rh_repo(
+        {
+            'product': PRDS['rhel'],
+            'repository-set': REPOSET['rhst7'],
+            'repository': REPOS['rhst7']['name'],
```
Comment (vsedmik) [inline tests/foreman/cli/test_activationkey.py:1443]: I would prefer single source of truth, all you need should be defined in `REPOS` already.
```suggestion
            'product': REPOS['rhst7']['product'],
            'repository-set': REPOS['rhst7']['reposet'],
            'repository': REPOS['rhst7']['name'],
```
Newer `rhst8` might work too.

**Review by vsedmik — CHANGES_REQUESTED** [review]
Looks good, just left a few comments. PRT pending.

File: tests/foreman/ui/test_activationkey.py [hunk 3]
```diff
@@ -1035,7 +1024,7 @@ def test_positive_host_associations(session, target_sat):
 @pytest.mark.tier3
 @pytest.mark.skipif((not settings.robottelo.repos_hosting_url), reason='Missing repos_hosting_url')
 def test_positive_service_level_subscription_with_custom_product(
-    session, function_entitlement_manifest_org, rhel7_contenthost, target_sat
+    session, function_sca_manifest_org, rhel7_contenthost, target_sat
 ):
     """Subscribe a host to activation key with Premium service level and with
     custom product
```
Comment (vsedmik) [inline tests/foreman/ui/test_activationkey.py:1030]: Update or remove `:steps:` 4, 7 and 8.

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k 'test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6706
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product --external-logging
Test Result : === 7 failed, 27 passed, 211 deselected, 604 warnings in 2231.37s (0:37:11) ====
```

File: tests/foreman/ui/test_activationkey.py [hunk 4]
```diff
@@ -611,11 +605,11 @@ def test_positive_fetch_product_content(target_sat, function_entitlement_manifes
         organization=org, repository=[rh_repo_id, custom_repo.id]
     ).create()
     cv.publish()
-    ak = target_sat.api.ActivationKey(content_view=cv, organization=org).create()
+    cvv = cv.read().version[0].read()
+    cvv.promote(data={'environment_ids': lce.id, 'force': True})
```
Comment (vijaysawant) [inline tests/foreman/ui/test_activationkey.py:609]: Use of `'force': True` is require during promotion?

**Review by vijaysawant — APPROVED** [review]
Ack
- IMHO, make `activation-key` tests SCA-compliant and in another PR, take remaning things like, use of global registration template method.
- I will leave up to author rather to include all changes in same PR or make separate PR for that.

File: tests/foreman/ui/test_activationkey.py (hunk 1 above)
Comment (shweta83) [reply tests/foreman/ui/test_activationkey.py:969]: There is an ongoing effort to replace these with global registration: https://github.com/SatelliteQE/robottelo/pull/14034
But please make the changes if possible and I can rebase my PR.

**Comment by vsedmik** [comment]
@ColeHiggins2 can you rebase? I think the GR is solved by https://github.com/SatelliteQE/robottelo/pull/14034 merged 3 days ago.

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k 'test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7034
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product --external-logging
Test Result : === 3 failed, 36 passed, 226 deselected, 1118 warnings in 4322.74s (1:12:02) ===
```

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k 'test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7046
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product --external-logging
Test Result : === 31 passed, 226 deselected, 935 warnings, 8 errors in 3778.73s (1:02:58) ====
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7072
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product --external-logging
Test Result : === 33 passed, 226 deselected, 987 warnings, 6 errors in 3464.05s (0:57:44) ====
```

**Review by vsedmik — CHANGES_REQUESTED** [review]
There are still some unresolved comments above and the CI/CQ fails with
```
  * Enumerated list ends without a blank line; unexpected unindent.

      :steps:
          1. Create a custom repo
    >     2. Create ak and add custom repo to ak
          4. Register a rhel6 chost with the ak
```
Can you address the comments and fix the CI/CQ?

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k 'test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7760
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product --external-logging
Test Result : = 3 failed, 33 passed, 228 deselected, 1039 warnings, 6 errors in 3831.60s (1:03:51) =
```

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k 'test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7766
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product --external-logging
Test Result : ========= 228 deselected, 272 warnings, 39 errors in 348.88s (0:05:48) =========
```

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k 'test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7771
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product --external-logging
Test Result : = 3 failed, 34 passed, 228 deselected, 1109 warnings, 5 errors in 4145.84s (1:09:05) =
```

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k 'test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product'

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k 'test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7856
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_activationkey.py tests/foreman/cli/test_activationkey.py tests/foreman/ui/test_activationkey.py -k test_positive_fetch_product_content or test_positive_create_with_name or test_positive_create_with_usage_limit_finite or test_positive_create_content_and_check_enabled or test_positive_update_usage_limit_to_finite_number or test_positive_update_usage_limit_to_unlimited or test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products or test_create_ak_with_syspurpose_set or test_update_ak_with_syspurpose_values or test_positive_copy_activationkey_and_check_content or test_positive_view_content_by_non_admin_user or test_positive_end_to_end_register or test_positive_update_rh_product or test_positive_add_rh_product or test_positive_add_custom_product or test_positive_add_rh_and_custom_products or test_positive_fetch_product_content or test_positive_add_multiple_aks_to_system or test_positive_service_level_subscription_with_custom_product --external-logging
Test Result : === 1 failed, 38 passed, 228 deselected, 1593 warnings in 6275.38s (1:44:35) ===
```

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_activationkey.py -k test_positive_add_custom_product

**Comment by ColeHiggins2** [comment]
Seems like a duplicate uuid made its way in to master causing code quality failures. I updated the test in repository.py to fix it.

**Review by vsedmik — APPROVED** [review]

**Pull request merged**